java-extensions/paimon-reader/src/main/java/com/starrocks/paimon/reader/PaimonSplitScanner.java

```diff
@@ -0,0 +1,214 @@
+
+package com.starrocks.paimon.reader;
+
+import com.starrocks.jni.connector.ColumnType;
+import com.starrocks.jni.connector.ColumnValue;
+import com.starrocks.jni.connector.ConnectorScanner;
+import com.starrocks.utils.loader.ThreadContextClassLoader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.paimon.catalog.Catalog;
+import org.apache.paimon.catalog.CatalogContext;
+import org.apache.paimon.catalog.CatalogFactory;
+import org.apache.paimon.catalog.Identifier;
+import org.apache.paimon.data.InternalRow;
+import org.apache.paimon.options.Options;
+import org.apache.paimon.reader.RecordReader;
+import org.apache.paimon.table.Table;
+import org.apache.paimon.table.source.ReadBuilder;
+import org.apache.paimon.table.source.Split;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.types.RowType;
+import org.apache.paimon.utils.InternalRowUtils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.options.CatalogOptions.METASTORE;
+import static org.apache.paimon.options.CatalogOptions.URI;
+import static org.apache.paimon.options.CatalogOptions.WAREHOUSE;
+
+public class PaimonSplitScanner extends ConnectorScanner {
+
+    private static final Logger LOG = LogManager.getLogger(PaimonSplitScanner.class);
+
+    private final String catalogType;
+    private final String metastoreUri;
+    private final String warehousePath;
+    private final String databaseName;
+    private final String tableName;
+    private final String splitInfo;
+    private final String[] requiredFields;
+    private ColumnType[] requiredTypes;
+    private DataType[] logicalTypes;
+    private Table table;
+    private RecordReader<InternalRow> reader;
+    private RecordReader.RecordIterator<InternalRow> batch;
+    private final int fetchSize;
+    private final ClassLoader classLoader;
+
+    public PaimonSplitScanner(int fetchSize, Map<String, String> params) {
+        this.fetchSize = fetchSize;
+        this.catalogType = params.get("catalog_type");
+        this.metastoreUri = params.get("metastore_uri");
+        this.warehousePath = params.get("warehouse_path");
+        this.databaseName = params.get("database_name");
+        this.tableName = params.get("table_name");
+        this.requiredFields = params.get("required_fields").split(",");
+        this.splitInfo = params.get("split_info");
+        this.classLoader = this.getClass().getClassLoader();
+        for (Map.Entry<String, String> kv : params.entrySet()) {
+            LOG.debug("key = " + kv.getKey() + ", value = " + kv.getValue());
+        }
+    }
+
+    private void initTable() throws IOException {
+        Options options = new Options();
+        options.setString(METASTORE.key(), catalogType);
+        options.setString(URI.key(), metastoreUri);
+        options.setString(WAREHOUSE.key(), warehousePath);
```

Comment:
Have you tested the hive catalog? Is it normal for be to access hms? Will there be permission issues?